.dockerignore

```diff
@@ -1,12 +1,45 @@
-/node_modules
-/dev
-/build
-/dist
-/.circleci
-/.git
-/.github
-/.vscode
-/packages/client/__generated__
-/.husky
-/.prettierrc
-/docs
+.git
```

Comment:
-1 instead of this file imperatively including the contents of `.gitignore` could we do a `cat .gitignore > .dockerignore` in the docker-build.yml below? that way we don't have to worry about making sure to update this file whenever we update `.gitignore`.

Reply:
This file isn't used only during the pipelines, it is also used if anyone builds the image locally, using any Dockerfile (some built by them or some of the ones we have). This file must always be kept up to date with the files that need to stay out of the Docker images produced by our repository, as a security measure.

I think your solution is enough for the pipeline, but do not work for the rest.

Reply:
can you tell me more about supporting local builds? I don't think any of the devs have a need to do that, so it might be a feature that we don't need to support.  if we do want to support it, could we concat the files in that dockerfile, too?

Reply:
gotcha! in this case I'm not sure if ignoring things like `/dist` is the right call because that would exclude all the built server assets, right?

could you provide some quick steps on how we should build this locally? With the `--no-deps` flag I imagine `/dist` is the _only_ thing we want to include in the build!

Reply:
Our Dockerfile (at least this one) do not use LOCAL files to build. That is the idea. Even if you build from your workstation, should not use already generated files.

When Docker starts to build, it takes the context as it is and start running the commands. If you check the lines 20 to 25 of the pipeline.dockerfile, you will see that the commands `yarn install --frozen-lockfile` and `yarn build` are executed. The dockerignore is there to ensure that no local /dist files are seen by the Docker daemon while building and it generates its own files, to be consistent between different docker builds done by different people in different machines using the same version of the code.

To be fair, this is only necessary while building on the workstations. Pipelines build are always done from scratch, so most of the .dockerignore isn't really useful. Probably we would be fine just with:

```
.git
.vscode
.github
.circleci
.husky
docs
```

More on [dockerignore](https://docs.docker.com/engine/reference/builder/#dockerignore-file)

There is no official way to specify a particular dockerignore file and that is why, for the pipeline, it must be on the root folder. Also for anyone building from local machine. And it must be then shared.

There are some [undocumented hacks](https://stackoverflow.com/questions/40904409/how-to-specify-different-dockerignore-files-for-different-builds-in-the-same-pr/57774684#57774684) we could test.